template.yaml

```diff
@@ -510,102 +576,124 @@ Resources:
       KmsKeyArn: '{{resolve:ssm:LambdaKmsKeyArn}}'
-      Policies:
-        - Statement:
-            - Sid: GetGlueTables
-              Effect: Allow
-              Action:
-                - glue:GetTable
-              Resource:
-                - !Sub arn:${AWS::Partition}:glue:${AWS::Region}:${AWS::AccountId}:catalog
-                - '{{resolve:ssm:AuditDataAnalysisGlueDatabaseArn}}'
-                - !Sub arn:${AWS::Partition}:glue:${AWS::Region}:${AWS::AccountId}:table/{{resolve:ssm:AuditDataAnalysisGlueDatabaseName}}/${AuditAnalysisTable}
-            - Sid: StartAthenaQuery
-              Effect: Allow
-              Action:
-                - athena:StartQueryExecution
-              Resource: '{{resolve:ssm:TicfAutomatedDataQueriesWorkgroupArn}}'
-            - Sid: ReadFromAnalysisBucket
-              Effect: Allow
-              Action:
-                - s3:ListBucket
-                - s3:GetObject
-              Resource:
-                - !GetAtt AnalysisBucket.Arn
-                - !Sub ${AnalysisBucket.Arn}/*
-            - Sid: WriteToResultsBucket
-              Effect: Allow
-              Action:
-                - s3:GetBucketLocation
-                - s3:GetObject
-                - s3:ListBucket
-                - s3:ListBucketMultipartUploads
-                - s3:ListMultipartUploadParts
-                - s3:AbortMultipartUpload
-                - s3:PutObject
-              Resource:
-                - '{{resolve:ssm:AthenaQueryOutputBucketArn}}'
-                - '{{resolve:ssm:AthenaQueryOutputBucketArn}}/*'
-            - Sid: Logs
-              Effect: Allow
-              Action:
-                - logs:CreateLogGroup
-                - logs:CreateLogStream
-                - logs:PutLogEvents
-              Resource: !GetAtt InitiateAthenaQueryLogs.Arn
-            - Sid: ReadSecrets
-              Effect: Allow
-              Action:
-                - secretsmanager:GetSecretValue
-              Resource: '{{resolve:ssm:ZendeskSecretSetArn}}'
-            - Sid: DecryptKmsKeys
-              Effect: Allow
-              Action:
-                - kms:Decrypt
-              Resource:
-                - '{{resolve:ssm:DatabaseKmsKeyArn}}'
-                - '{{resolve:ssm:SecretsKmsKeyArn}}'
-                - '{{resolve:ssm:SqsKmsKeyArn}}'
-            - Sid: QueryRequestTableReadWrite
-              Effect: Allow
-              Action:
-                - dynamodb:PutItem
-                - dynamodb:GetItem
-                - dynamodb:UpdateItem
-              Resource: '{{resolve:ssm:QueryRequestTableArn}}'
-            - Sid: AthenaOutputEncryptKmsKey
-              Effect: Allow
-              Action:
-                - kms:GenerateDataKey
-                - kms:Encrypt
-                - kms:Decrypt
-              Resource: '{{resolve:ssm:AthenaQueryOutputBucketKmsKeyArn}}'
-            - Sid: Events
-              Effect: Allow
-              Action:
-                - events:PutEvents
```

Comment:
This permission `PutEvents` was here originally, but when I moved it out of the inline policy, I got a Checkov warning for being too permissive (because of the `*`). I couldn't see what it was for, and all the integration tests pass without it being there, so I don't think we need it. Possibly something that was added in an early development phase that somebody thought was fixing something, but it actually wasn't.